pkg/analytics/client.go

```diff
@@ -0,0 +1,232 @@
+		return newClientID(), nil
+	}
+
+	return ID, nil
+}
+
+// Generates a new unique id
+func newClientID() string {
+	return uuid.New().String()
+}
+
+// enableGoogleAnalytics enable google analytics for HPE CLI
+func enableGoogleAnalytics() bool {
+	var disableGA string
+
+	d, err := db.Open()
+
+	if err != nil {
+		logrus.Debug("Unable to open DB to get DisableAnalyticsKey")
+
+		return false
+	}
+	defer d.Close()
+
+	if err := d.Get(DisableAnalyticsKey, &disableGA); err != nil {
+		return true
+	}
+
+	if disableGA == "true" {
+		logrus.Debugf("Found existing DisableAnalyticsKey, updating it's value to : %s", disableGA)
+
+		err := d.Put(DisableAnalyticsKey, "false")
+		if err != nil {
+			logrus.Debugf("Unable to put the key %s in to DB", DisableAnalyticsKey)
+			return false
+		}
+
+		return true
+	} else if disableGA == falseValue {
+		return true
+	}
+
+	return false
+}
+
+// disableGoogleAnalytics disable google analytics for HPE CLI
+func disableGoogleAnalytics() bool {
+	var disableGA string
+
+	d, err := db.Open()
+
+	if err != nil {
+		logrus.Debug("Unable to open DB to get GAClientIDKey ")
+		return false
+	}
+	defer d.Close()
+
+	if err := d.Get(DisableAnalyticsKey, &disableGA); err != nil {
+		logrus.Debugf("Didn't find existing DisableAnalyticsKey, creating a new one: %s", DisableAnalyticsKey)
+
+		err := d.Put(DisableAnalyticsKey, "true")
+		if err != nil {
+			logrus.Debugf("Unable to put the key %s in to DB", DisableAnalyticsKey)
+			return false
+		}
+
+		return true
+	}
+
+	if disableGA == falseValue {
+		logrus.Debugf("Didn't find existing enableAnalytics, creating a new one: %s", DisableAnalyticsKey)
+		_ = d.Put(DisableAnalyticsKey, "true")
+
+		return true
+	} else if disableGA == "true" {
+		return true
+	}
+
+	return false
+}
+
+// CheckGoogleAnalytics Check whether google analytics is enabled or disabled in db
+func CheckGoogleAnalytics() bool {
+	var disableGA string
+
+	d, err := db.Open()
+
+	if err != nil {
+		logrus.Debug("Unable to open DB to get GAClientIDKey ")
+		return false
+	}
+	defer d.Close()
+
+	if err := d.Get(DisableAnalyticsKey, &disableGA); err != nil {
+		logrus.Debugf("Didn't found existing DisableAnalyticsKey key %s", disableGA)
+
+		return true
+	}
+
+	if disableGA == falseValue {
+		return true
+	}
+
+	return false
+}
+
+// TrackEvent Measurement Protocol api to track user events
+func (c *Client) TrackEvent() (string, error) {
+	const uriPath = "/collect"
+
+	const host = "https://www.google-analytics.com"
+
+	u, err := url.Parse("https://www.google-analytics.com/collect")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	u.Scheme = "https"
+	u.Host = "google-analytics.com"
+	u.Path = "/collect"
+
+	q := u.Query()
+	q.Set("v", "1")
+	q.Set("tid", TrackingID)
+	q.Set("t", c.EventHitType)
+	q.Set("ea", c.EventAction)
+	q.Set("ev", c.EventValue)
+	q.Set("ec", c.Eventcategory)
+	q.Set("ua", c.UserAgent)
+	q.Set("an", c.ApplicationName)
+	q.Set("av", c.ApplicationVersion)
+
+	id, err := clientID()
```

Comment:
maybe do this check at the beginning of the method?  We don't need to do all of the work in this method if we don't have a client ID